src/MatrixClient.cc

```diff
@@ -464,6 +464,45 @@ MatrixClient::getOwnProfile() noexcept
 }
 
 void
+MatrixClient::getOwnCommunities() noexcept
+{
+        QUrlQuery query;
+        query.addQueryItem("access_token", token_);
+
+        QUrl endpoint(server_);
+        endpoint.setPath(clientApiUrl_ + "/joined_groups");
+        endpoint.setQuery(query);
+
+        QNetworkRequest request(QString(endpoint.toEncoded()));
+
+        QNetworkReply *reply = get(request);
+        connect(reply, &QNetworkReply::finished, this, [this, reply]() {
+                reply->deleteLater();
+
+                int status = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+
+                if (status >= 400) {
+                        qWarning() << reply->errorString();
+                        return;
+                }
+
+                auto data = reply->readAll();
+                auto json = QJsonDocument::fromJson(data).object();
+                try {
+                        QList<QString> response;
+                        for (auto it = json["groups"].toArray().constBegin();
+                             it != json["groups"].toArray().constEnd();
+                             it++) {
+                                response.append(it->toString());
```

Comment:
I get a SEGV here when accessing `it.operator->()`. While debugging I found that the `...toArray().constEnd()` invalidates the iterator. When using a single reference to the array then it works fine.